src/services/download/s3.rs

```diff
@@ -78,15 +78,14 @@ impl S3Downloader {
 
     pub async fn download_source(
         &self,
-        source: Arc<S3SourceConfig>,
-        download_path: SourceLocation,
+        file_source: S3ObjectFileSource,
```

Comment:
Would it be possible to keep those attributes separate in the downloader implementations? There's probably a good reason you combined them, but looking at the result now I think the previous signature was more straight forward and required less type definitions.

Reply:
I found this easier to work with to be honest.  Plus it allows to implement things like `.url()` on the ObjectFileSource type instead of needing them to be free-standing functions having knowledge about this type.

So in short, I'd like to keep it like this (unsurprisingly).

Reply:
Understood, and this is probably very subjective. Feel free to merge like this 👍  

It would not look as strange to me if the downloaders defined their types in the same file rather than all source types 
(+ configs) being defined in a single file. That way, at least a downloader is self contained, and everything composes that.

The interface change to the downloaders is very unfortunate. Passing config + location in a single argument seems like an additional indirection when looking at it in isolation. As an idea, would it be possible to destructure this on a higher level and at least keep `download/*`the same?

Reply:
In summary, going to leave this.  Having them as two separate structs meant there was a lot of random guessing how two of these bits went together.  This way brings more type safety.  (it's a bit unfortunate that we use an enum rather than a trait for the overarching type, but I don't want to make everything that handles this generic just now)